tasks/build.js

```diff
@@ -0,0 +1,70 @@
+const rollup = require('rollup')
+const resolve = require('rollup-plugin-node-resolve')
+const babel = require('rollup-plugin-babel')
+const commonjs = require('rollup-plugin-commonjs')
+const json = require('rollup-plugin-json')
+const fileSize = require('rollup-plugin-filesize')
+const rimraf = require('rimraf')
+const chalk = require('chalk')
+const path = require('path')
+
+function getConfigFile() {
+  try {
+    return require(path.join(process.cwd(), 'bundle.config.js'))
+  } catch {
+    console.log(
+      chalk.red(`bundle.config.js file is missing for ${process.cwd()}`)
+    )
+    console.log(chalk.red('build bundle failed'))
+    process.exit()
+  }
+}
+
+async function asyncRimRaf(filepath) {
+  return new Promise((resolve, reject) =>
+    rimraf(filepath, error => {
+      if (error) {
+        reject(error)
+        return
+      }
+      resolve()
+    })
+  )
+}
+
+async function createBundle(bundle, format) {
+  const inputOptions = {
+    input: `${bundle.entry}`,
+    plugins: [
+      babel({
+        exclude: 'node_modules/**'
+      }),
+      resolve(),
+      commonjs(),
+      json(),
+      fileSize()
+    ],
+    external: bundle.external
+  }
+
+  const outputOptions = {
+    file: `dist/index.${format}.js`,
+    format: format
+  }
+
+  const result = await rollup.rollup(inputOptions)
+  await result.write(outputOptions)
+}
+
+async function build(bundleLabel) {
```

Comment:
What does having a custom script do beyond what the Rollup CLI can do? Personally I'd rather see a static config file that is consumed by Rollup. Rimraf could be handled with an npm script, but I'm not seeing anything else that requires the separate node script

Reply:
At this point, everything could be done with rollup config file.  Personally, I think there are 3 major benefits of this.

1. it's easier to add the global level configuration.  Like the sharable plugins.  Yes, it could be done via import a root level `rolluo.config.js` file, but it looks in each package, it still requires a lot of changes to that import. Just not clean when the bundles are getting complicated.

2. Easier to extend. Now it's just the Rimraf function in addition to rollup, but there could be more.  https://github.com/facebook/react/tree/master/scripts/rollup could show where this pattern will go.

3. With a bundle script, to a point where we are switching to another tool, it will be easier. 

However, it really depends how complex the DS will go. Since I have no idea of that, I'm using something I think will be useful in that situation.  At this point, there is no obvious gain from it. So if you think rollup config file could be good to handle most our use cases, I'm good with that too.

Reply:
Thanks for the explanation! Might be worth digging into point #1 a little deeper – maybe in the weekly meeting.

> it really depends how complex the DS will go

As I tend to think generally in software development, I think we should solve for the problems we have now, not the problems that we *might have* in the future, if that makes sense

Reply:
definitely, since you have a much clearer vision of DS, I trust you in that